source/fem-dem/gls_sharp_navier_stokes.cc

```diff
@@ -142,28 +146,39 @@ GLSSharpNavierStokesSolver<dim>::refine_ib()
           cell->get_dof_indices(local_dof_indices);
           for (unsigned int p = 0; p < particles.size(); ++p)
             {
-              unsigned int count_small     = 0;
-              Point<dim>   center_immersed = particles[p].position;
-
+              unsigned int   count_small     = 0;
+              Point<dim>     center_immersed = particles[p].position;
+              Tensor<1, dim> r;
+              r[0] = particles[p].radius;
+
+              // Check if a point on the boundary is contained in that cell.
+              // The idea here is that if the particle is much smaller than the
+              // cell, it can be hard to detect it, so we refine this cell
+              // anyway if it contains this random point on the boundary. This
+              // will help the detection of the particle on the next refinement.
+              bool cell_as_ib_inside =
+                cell->point_inside(particles[p].position + r);
               for (unsigned int j = 0; j < local_dof_indices.size(); ++j)
                 {
                   // Count the number of dofs that are smaller or larger than
                   // the radius of the particles if all the dof are on one side
```

Comment:
```suggestion
                  // the radius of the particles if all the dofs are on one side
```
I didn't understand what you mean by number of dofs that are smaller or larger than
 the radius of the particles